packages/locale/src/TranslatedString.tsx

```diff
@@ -1,16 +1,29 @@
 import React, { FunctionComponent } from 'react';
 
-import withLanguage, { WithLanguageProps } from './withLanguage';
+import getLanguageService from './getLanguageService';
+import LocaleContext, { LocaleContextType, useLocale } from './LocaleContext';
 
 export interface TranslatedStringProps {
     id: string;
     data?: any;
 }
 
-const TranslatedString: FunctionComponent<TranslatedStringProps & WithLanguageProps> = ({
-    data,
-    id,
-    language,
-}) => <>{language.translate(id, data)}</>;
+const TranslatedStringWithContext: FunctionComponent<TranslatedStringProps> = ({ data, id }) => {
+    const { language } = useLocale();
 
-export default withLanguage(TranslatedString);
+    return <>{language.translate(id, data)}</>;
+};
+
+const TranslatedString: FunctionComponent<TranslatedStringProps> = (props) => {
+    const localeContextValue: LocaleContextType = {
+        language: getLanguageService(),
+    };
+
+    return (
+        <LocaleContext.Provider value={localeContextValue}>
```

Comment:
Is there any special reason for putting `<LocaleContext.Provider value={localeContextValue}>` here?
We have `useLocale();` to check if this component is inside a `localeContext`, hence here we are applying another layer of the same context.

Reply:
@bc-peng To test the changes for this requirement we only ran the unit tests. Now if we don't wrap it with the provider then several tests were failing. 
<img width="776" alt="Screenshot 2023-09-25 at 7 15 56 pm" src="https://github.com/bigcommerce/checkout-js/assets/141614330/73452635-5d57-4259-b6a5-a57c6a0f1bda">

I added a provider to this file because the provider just needed language service which is common throughout and does not change.

What do you suggest? All those tests should be wrapped with the provider? And how do we make sure all those integration packages will work fine without wrapping `TranslatedString` with provider?

Reply:
We have applied the locale context at the very top.
https://github.com/bigcommerce/checkout-js/blob/a255f2ab02ef0178b3bc0bb2b0977829625f8b73/packages/core/src/app/checkout/CheckoutApp.tsx#L59

e2e tests inside each integration package ensure the package will work fine.

I think you are adding the context again for the testing purpose, could we find out how many tests will be impacted here if we don't do so?

Reply:
<img width="381" alt="Screenshot 2023-09-26 at 1 57 05 pm" src="https://github.com/bigcommerce/checkout-js/assets/141614330/82e70f26-9406-4e4c-b847-25a4cab34b53">

@bc-peng I have ran unit tests here `npm run test:core` and we have 52 failures there.